includes/providers/gam/class-gam-scripts.php

```diff
@@ -158,11 +158,27 @@ function( $a, $b ) {
 		$prepared_unit_data = apply_filters( 'newspack_ads_gtag_ads_data', $prepared_unit_data );
 
 		do_action( 'newspack_ads_gtag_before_script', $ad_config, $prepared_unit_data );
+
+		ob_start();
+		self::output_frontend_js( $ad_config, $prepared_unit_data, $common_targeting );
+		echo apply_filters( 'newspack_ads_frontend_js', ob_get_clean() ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+
+		do_action( 'newspack_ads_gtag_after_script', $ad_config, $prepared_unit_data );
+	}
+
+	/**
+	 * Output frontend JS.
+	 *
+	 * @param array $ad_config Ad config.
+	 * @param array $prepared_unit_data Ad unit data.
+	 * @param array $common_targeting Common targeting.
+	 */
+	private static function output_frontend_js( $ad_config, $prepared_unit_data, $common_targeting ) {
```

Comment:
```suggestion
	/**
	 * Print GPT/GAM script.
	 *
	 * @param array $ad_config          Ad config.
	 * @param array $prepared_unit_data Ad unit data.
	 * @param array $common_targeting   Common targeting.
	 */
	private static function print_gpt_script( $ad_config, $prepared_unit_data, $common_targeting ) {
```

NIT: The parent method is called `insert_gpt_header_script`, we should follow a related naming.